scrape/scrape.go

```diff
@@ -1503,6 +1517,17 @@ loop:
 		if ok {
 			ref = ce.ref
 			lset = ce.lset
+
+			// Update metadata only if it's changed in the current iteration.
```

Comment:
```suggestion
			// Update metadata only if it changed in the current iteration.
```

Reply:
Sorry for the force push; I initially tried quickly commiting through the Github UI but it failed the DCO check.